From fdcc6e69c64f8971d225c69890ec1dfe553fe613 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:09 -0400
Subject: [tools] perf/record: Use perf_data_file__write for output file

Message-id: <1408953210-26343-182-git-send-email-jolsa@redhat.com>
Patchwork-id: 88852
O-Subject: [PATCH RHEL7.1 BZ1133083 181/322] perf record: Use perf_data_file__write for output file
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 50a9b8680613a2708ca882d982dcfa4fd9a66673
Author: Jiri Olsa <jolsa@redhat.com>
Date: Fri Nov 22 13:11:24 2013 +0100

description
===========
Changing the file output code to use the newly
added perf_data_file__write interface.

No functional change intended.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index c1c1200..8eed3d7 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -76,24 +76,19 @@ struct perf_record {
  long   samples;
 };
 
-static int perf_record__write(struct perf_record *rec, void *buf, size_t size)
+static ssize_t perf_record__write(struct perf_record *rec,
+      void *buf, size_t size)
 {
- struct perf_data_file *file = &rec->file;
-
- while (size) {
-  ssize_t ret = write(file->fd, buf, size);
-
-  if (ret < 0) {
-   pr_err("failed to write perf data, error: %m\n");
-   return -1;
-  }
-
-  size -= ret;
-  buf += ret;
+ struct perf_session *session = rec->session;
+ ssize_t ret;
 
-  rec->bytes_written += ret;
+ ret = perf_data_file__write(session->file, buf, size);
+ if (ret < 0) {
+  pr_err("failed to write perf data, error: %m\n");
+  return -1;
  }
 
+ rec->bytes_written += ret;
  return 0;
 }
 
-- 
1.7.1